src/dimensions.js

```diff
@@ -74,8 +80,19 @@ export function Dimensions(scales, geometry, axes, options = {}) {
   };
 }
 
-function autoHeight({y, fy, fx}, geometry) {
+function autoHeight({y, fy, fx}) {
   const nfy = fy ? fy.scale.domain().length : 1;
-  const ny = y ? (isOrdinalScale(y) ? y.scale.domain().length : Math.max(7, 17 / nfy)) : geometry ? 17 : 1;
-  return !!(y || fy || geometry) * Math.max(1, Math.min(60, ny * nfy)) * 20 + !!fx * 30 + 60;
+  const ny = y ? (isOrdinalScale(y) ? y.scale.domain().length : Math.max(7, 17 / nfy)) : 1;
+  return !!(y || fy) * Math.max(1, Math.min(60, ny * nfy)) * 20 + !!fx * 30 + 60;
+}
+
+function autoHeightProjection({fy, fx}, ratio, width, marginY) {
+  if (ratio > 0) {
+    return Math.round(
+      Math.max(
+        140,
+        Math.min(1260, (width * (fx ? fx.scale.bandwidth() : 1) * ratio) / (fy ? fy.scale.bandwidth() : 1))
```

Comment:
I think this depends on us not having set the (automatic) range of the fx/fy scales here. It’d be better to not depend on the order of operations and instead look at the scale’s padding.

Reply:
fixed